### PR TITLE
Changes to conftest.py for translate tests

### PR DESCRIPTION
**Pull request by fmalatino**
**Description**
The conftest.py file for the translate tests has been changed to match centralized test structure in NDSL.

**How Has This Been Tested?**
Tested with the current translate test structure triggering the appropriate downstream repositories.
**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] Targeted model if this changed was triggered by a model need/shortcoming

### CODE REVIEW

**Review by FlorianDeconinck — DISMISSED** [review]
LGTM. Fixed post NDSL change.

**Review by oelbert — DISMISSED** [review]
👍 Good to merge once NDSL is in

**Comment by fmalatino** [comment]
Ran into a bit of a circular issue as the failure in the pace unit tests is due to a change included in this PR, moving `mysign` to an external variable.

**Review by bensonr — DISMISSED** [review]
based on what I know, this looks good to me

File: .github/workflows/translate.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+name: "Translate test (subset)"
```
Comment (mlee03) [inline .github/workflows/translate.yml]: this file is not needed

Comment (fmalatino) [reply .github/workflows/translate.yml]: Removed.

**Review by bensonr — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -15,8 +15,8 @@
     "xarray",
 ]
 
-test_requirements = ["pytest", "pytest-subtests", "serialbox"]
-ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@2024.04.00"]
+test_requirements = ["pytest", "pytest-subtests", "serialbox", "coverage"]
+ndsl_requirements = ["ndsl @ git+https://github.com/NOAA-GFDL/NDSL.git@develop"]
```
Comment (mlee03) [inline setup.py:19]: quick question, why the develop branch and not a specific tag?

Comment (fmalatino) [reply setup.py:19]: We are using the `develop` branch as the release is not ready yet, and that the `develop` branch contains the configurations needed for testing. Once the orchestration issue is resolved the new release will be made and there will be a subsequent PR to update this.

**Review by mlee03 — APPROVED** [review]

**Pull request merged**